core/src/main/java/com/sequenceiq/distrox/v1/distrox/service/upgrade/DistroXUpgradeAvailabilityService.java

```diff
@@ -60,14 +60,46 @@ public boolean isRuntimeUpgradeEnabled(String userCrn) {
         }
     }
 
+    public boolean isRuntimeUpgradeEnabledByAccountId(String accountId) {
+        return entitlementService.datahubRuntimeUpgradeEnabled(accountId);
+    }
+
     public UpgradeV4Response checkForUpgrade(NameOrCrn nameOrCrn, Long workspaceId, UpgradeV4Request request, String userCrn) {
-        verifyRuntimeUpgradeEntitlement(userCrn, request);
         Stack stack = stackService.getByNameOrCrnInWorkspace(nameOrCrn, workspaceId);
         String accountId = Crn.safeFromString(userCrn).getAccountId();
         UpgradeV4Response response = stackOperations.checkForClusterUpgrade(accountId, stack, workspaceId, request);
-        List<ImageInfoV4Response> filteredCandidates = filterCandidates(accountId, stack, request, response);
-        response.setUpgradeCandidates(filteredCandidates);
-        return response;
+        UpgradeV4Response filteredResponse =
+                filterOnlyPatchUpgradesIfRuntimeUpgradeDisabled(accountId, nameOrCrn, request, response);
+        List<ImageInfoV4Response> filteredCandidates = filterCandidates(accountId, stack, request, filteredResponse);
+        filteredResponse.setUpgradeCandidates(filteredCandidates);
+        return filteredResponse;
+    }
+
+    private UpgradeV4Response filterOnlyPatchUpgradesIfRuntimeUpgradeDisabled(String accountId, NameOrCrn nameOrCrn, UpgradeV4Request upgradeV4Request,
+            UpgradeV4Response upgradeV4Response) {
+        if (!isRuntimeUpgradeEnabledByAccountId(accountId)) {
+            UpgradeV4Response filteredUpgradeResponse =
+                    new UpgradeV4Response(upgradeV4Response.getCurrent(), upgradeV4Response.getUpgradeCandidates(), upgradeV4Response.getReason());
+            if (upgradeV4Response.getCurrent() != null) {
+                List<ImageInfoV4Response> upgradeCandidates = filteredUpgradeResponse.getUpgradeCandidates();
+                String currentCdpVersion = filteredUpgradeResponse.getCurrent().getComponentVersions().getCdp();
```

Comment:
this should use upgradeV4Response instead of filteredUpgradeResponse as the latter is for the filtered results